### ISSUE TITLE
Error Reporting for Ultra-High Definition Images

### ISSUE THREAD

**Issue by yangboning** [ultralytics/ultralytics#4681]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have the need to detect small targets in a large scene while using YOLOv8.
For this I processed some footage and images:
Enhancement of a 1220KB image to 43909KB using Real-ESRGAN        Resolution enhancement from 3840x2160 to 15360x8640

Accordingly I quadrupled the coordinates in the labeled json file.

Then it reported an error when feeding data to the model:
![image](https://github.com/ultralytics/ultralytics/assets/37468757/f4e8fb13-33a0-4ad1-bb41-9316f60fa647)
![image](https://github.com/ultralytics/ultralytics/assets/37468757/69ad7746-3489-401f-90bb-66a3962cb6d1)

### Additional

Here's the json file comparison:
![image](https://github.com/ultralytics/ultralytics/assets/37468757/958e9121-b471-4a03-b546-d586a5a4292b)
Here's a comparison of the YOLO files:
![image](https://github.com/ultralytics/ultralytics/assets/37468757/0f82800c-0d28-4fcf-b75e-518fc7a6c538)

**Comment by yangboning** [comment]
Here's the json file comparison:
![image](https://github.com/ultralytics/ultralytics/assets/37468757/958e9121-b471-4a03-b546-d586a5a4292b)
Here's a comparison of the YOLO files:
![image](https://github.com/ultralytics/ultralytics/assets/37468757/0f82800c-0d28-4fcf-b75e-518fc7a6c538)

**Comment by yangboning** [comment]
The current scheme intercepts the detection target from the enhanced image again. Then re-labeling makes the image smaller. But this will lose some scene information.

**Comment by glenn-jocher** [comment]
@yangboning hello,

Thanks for providing such detailed information about the issue you're encountering with ultra-high-definition images in YOLOv8. Your approach of enhancing the input image resolution and adjusting the target coordinates in the JSON file appears correct in principle.

From the error messages you've provided, it looks like you may be encountering a resource limitation issue. Ultra-high-definition images, like the ones you're using, require significantly more computational resources to process, which may lead to memory errors or other resource-related issues, particularly if you're using a machine with limited computational power.

It's worth noting, however, that smaller images don't necessarily lose scene information, particularly if the objects you want to detect are still identifiable at the smaller resolution. The YOLOv8 model is highly efficient at detecting objects across various scales and resolutions. 

As for retaining the full scene context, you could consider breaking your high-resolution images into smaller tiles, perform object detection on these tiles, and then reconstruct the full scene with the detected objects afterwards.

Remember, YOLOv8 has intrinsic multi-scale detection capabilities, which can recognize small objects in large scenes, given that the objects are properly represented in the training data.

I hope this gives you some guidance on how to proceed. If you continue to experience issues or have any more questions, feel free to get back in touch.

Best Regards,
Glenn Jocher

**Comment by yangboning** [comment]
@glenn-jocher hello,
First of all, thank you for your guidance.
Next I'll provide some detailed hardware information and actual issues for tracking and feedback purposes.
Hardware:
![image](https://github.com/ultralytics/ultralytics/assets/37468757/df97cec8-d83e-4a18-b835-d4ead3fc8252)
![image](https://github.com/ultralytics/ultralytics/assets/37468757/9d5d9b49-3b1d-4369-b698-af68b88c4625)
![image](https://github.com/ultralytics/ultralytics/assets/37468757/29e89f9c-c64d-472a-9047-ce5315494636)
Power：USCORSAIR SF750

![image](https://github.com/ultralytics/ultralytics/assets/37468757/9898f678-e922-4598-9bbe-304953bba311)
![image](https://github.com/ultralytics/ultralytics/assets/37468757/6ca50557-a35e-4215-a447-12a8b035def5)
As you can see in the graph above, the accuracy of sp and hp is very low, both below 0.5. There are 107 labeled images for training. 
!yolo task=detect mode=train epochs=900 data=data_custom.yaml model=yolov8m.pt imgsz=640

However the training does not reach 900 this iteration and stops at 100+.
I have now changed imgsz=640 to 1280. still training at the moment and this time the training is getting very long as shown.
![image](https://github.com/ultralytics/ultralytics/assets/37468757/b8e08462-8795-484c-a888-a92d14101d2b)

**Comment by yangboning** [comment]
Using Real-ESRGAN makes the picture sharper, that's just how I feel as a human. Even though the image ends up being very blurry after compression and transformation in the convolution, I haven't tried to see if changing the sharpness of the image from the input side will have an effect on the final trained model.
Release the effect image:
Original Image：
![image](https://github.com/ultralytics/ultralytics/assets/37468757/5cf3c6eb-b0dd-4cde-a003-61d9d366b7c9)
Enhanced image：
![image](https://github.com/ultralytics/ultralytics/assets/37468757/c1c20de4-238d-41fc-b7cd-8c49f2d46e6c)

**Comment by agjunyent** [comment]
As far as I know, the yolo files with the boxes are in relative space, so it makes absolutely no sense to multiply them x4 too, as it does not matter the scale of the image, the relative coordinates are always going to be the same.

Do NOT scale them up, I see you have numbers >1, and that's invalid

**Comment by glenn-jocher** [comment]
Hello @yangboning,

Thank you for pointing this out, and you're absolutely correct. The coordinates provided in the YOLO files are relative to the dimensions of the image, not the absolute pixel values. Consequently, they should always remain within the interval [0,1] no matter how much you scale or resize your images. 

If you're indeed scaling up the coordinates and obtaining values greater than 1, this will most likely produce invalid results and may even cause the model to fail or behave unexpectedly. Therefore, as you've suggested, it's vital to ensure that the scaled coordinates remain within the valid range regardless of the image's resolution.

Be sure to keep this information in mind as you continue to work on your project.

Best Regards,
Glenn Jocher
